src/NextGen/Framework/FormExtensions/Registrars/FormExtensionRegistrar.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace Give\NextGen\Framework\FormExtensions\Registrars;
+
+use Give\Framework\Exceptions\Primitives\InvalidArgumentException;
+use Give\Log\Log;
+use Give\NextGen\Framework\FormExtensions\Exceptions\OverflowException;
+use Give\NextGen\Framework\FormExtensions\FormExtension;
+
+/**
+ * @unreleased 
+ */
+class FormExtensionRegistrar
+{
+    /**
+     * @var array
+     */
+    protected $extensions = [];
+
+    /**
+     * @unreleased 
+     */
+    public function getExtensions(): array
+    {
+        return $this->extensions;
+    }
+
+    /**
+     * @unreleased 
+     *
+     * @throws InvalidArgumentException
+     */
+    public function getExtension(string $id): FormExtension
+    {
+        if (!$this->hasExtension($id)) {
+            throw new InvalidArgumentException("No extension exists with the ID {$id}");
+        }
+
+        /** @var FormExtension $extension */
+        $extension = give($this->extensions[$id]);
+
+        return $extension;
+    }
+
+    /**
+     * @unreleased 
+     */
+    public function hasExtension(string $id): bool
+    {
+        return isset($this->extensions[$id]);
+    }
+
+    /**
+     * @unreleased 
+     */
+    public function registerExtension(string $extensionClass)
+    {
+        try {
+            $this->register($extensionClass);
+        } catch (InvalidArgumentException $invalidArgumentException) {
+            Log::error('Form Extension Registration', ['data' => $invalidArgumentException->getMessage()]);
+            throw $invalidArgumentException;
+        } catch (OverflowException $overflowException) {
+            Log::error('Form Extension Registration ', ['data' => $overflowException->getMessage()]);
```

Comment:
These log messages should probably be more specific, explaining what went wrong.